src/vmsa_list.rs

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (C) 2022 Advanced Micro Devices, Inc.
+ *
+ * Authors: Carlos Bilbao <carlos.bilbao@amd.com> and
+ *          Tom Lendacky <thomas.lendacky@amd.com>
+ */
+
+use crate::funcs;
+use crate::locking::LockGuard;
+use crate::locking::SpinLock;
+
+use alloc::vec::Vec;
+use lazy_static::lazy_static;
+use x86_64::addr::PhysAddr;
+
+#[derive(Clone, Copy, Debug)]
+struct VmsaInfo {
+    gpa: u64,
+    apic_id: u32,
+}
+
+#[allow(dead_code)]
+impl VmsaInfo {
+    funcs!(gpa, u64);
+    funcs!(apic_id, u32);
+}
+
+lazy_static! {
+    static ref VMSA_LIST: SpinLock<Vec<VmsaInfo>> = SpinLock::new(Vec::with_capacity(512));
+}
+
+pub fn del_vmsa(gpa: PhysAddr) -> bool {
```

Comment:
Since this is now public, lets rename this to vmsa_list_remove or vmsa_list_delete.

Reply:
I tried only to move code and not change it at the same time. 
Should I rename the methods in a subsequent commit inside this PR?

If we're changing, WDYT about a `struct VmsaList` with several methods on it, like `add`, `delete`, `is_present`, etc (or maybe use the `Vec` method names: `push`, `remove`, `contains`). 

The static-global variable can now be of type `VmsaList` (and the `SpinLock<Vec<VmsaInfo>>` will be its field). The callsites will look like: `VMSA_LIST.delete(...)` (instead of `vmsa_list_delete(...)`).

Reply:
Yes, I like that.

Reply:
done.